lizrd/grid/infrastructure.py

```diff
@@ -0,0 +1,279 @@
+import abc
+import os
+import platform
+
+
+class MachineBackend(abc.ABC):
+    @classmethod
+    @abc.abstractmethod
+    def get_common_directory(cls) -> str:
+        pass
+
+    @classmethod
+    @abc.abstractmethod
+    def get_cache_path(cls) -> str:
+        pass
+
+    @classmethod
+    def get_singularity_image(cls) -> str:
+        image_name = "sparsity_2024.02.06_16.14.02.sif"
+        common_dir = cls.get_common_directory()
+        return f"{common_dir}/images/{image_name}"
+
+    @classmethod
+    @abc.abstractmethod
+    def get_grid_entrypoint(cls) -> str:
+        pass
+
+    @classmethod
+    def get_default_train_dataset_path(cls, dataset_type: str):
+        return None
+
+    @classmethod
+    def get_default_validation_dataset_path(cls, dataset_type: str):
+        return None
+
+    @classmethod
+    def get_cluster_default_params(cls, dataset_type) -> dict:
+        return {
+            "train_dataset_path": cls.get_default_train_dataset_path(dataset_type),
+            "validation_dataset_path": cls.get_default_validation_dataset_path(
+                dataset_type
+            ),
+            "common_directory": cls.get_common_directory(),
+            "hf_datasets_cache": cls.get_cache_path(),
+            "singularity_image": cls.get_singularity_image(),
+            "grid_entrypoint": cls.get_grid_entrypoint(),
+        }
+
+    @classmethod
+    def prepare_default_infrastructure_params(cls, dataset_type: str):
+        infrastructure_params_dict = COMMON_DEFUALT_INFRASTRUCTURE_ARGS
+        cluster_default_arg_dict = cls.get_cluster_default_params(dataset_type)
+        infrastructure_params_dict.update(cluster_default_arg_dict)
+        return infrastructure_params_dict
+
+    @classmethod
+    @abc.abstractmethod
+    def get_subprocess_args(
+        cls,
+        slurm_command,
+        setup_args,
+        training_args,
+        singularity_env_arguments,
+        singularity_mount_paths,
+        runner_params,
+    ):
+        pass
+
+
+class AthenaBackend(MachineBackend):
+    @classmethod
+    def get_common_directory(cls) -> str:
+        return "/net/pr2/projects/plgrid/plggllmeffi"
+
+    @classmethod
+    def get_cache_path(cls) -> str:
+        return f"/net/tscratch/people/{os.environ.get('USER')}/.cache"
+
+    @classmethod
+    def get_grid_entrypoint(cls) -> str:
+        return "lizrd/scripts/grid_entrypoint.sh"
+
+    @classmethod
+    def get_subprocess_args(
+        cls,
+        slurm_command,
+        setup_args,
+        training_args,
+        singularity_env_arguments,
+        singularity_mount_paths,
+        runner_params,
+    ):
```

Comment:
Po przerzuceniu funkcji budujących te argumenty do klasy, można je będzie zbudować tutaj
```suggestion
        runner_params,
    ):
        singularity_mount_paths = self.make_singularity_mount_paths(...)
        singularity_env_arguments = self.make_singularity_env_arguments(...)
```

Reply:
nie mam `neptune_key` ani `wandb_key` więc tylko mount_paths moge przerzucić